### PR TITLE
new starter template, based on cherry picked commits from VVB work

### PR DESCRIPTION
**Pull request by jegasriskantha**
## Summary: 

- new starter template based on VVB work 
- includes sidebar, auth and breadcrumbs

### CODE REVIEW

**Comment by mattcorner** [comment]
It would be nicer if 404 was wrapped in Layout so that we can navigate through the app.

image

**Comment by mattcorner** [comment]
On 404 my favicon reverts to remix
image

**Comment by mattcorner** [comment]
Documentation link goes nowhere. Should we seed with an empty docusaurus?

**Comment by mattcorner** [comment]
Even though I am on the example page, I get old tab headers. I don't think we've configured it to set anything, at the very least it should be the app name from the package.json as a default.

image

File: app/src/Layout.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  // use form submission hook
+  const submit = useSubmit();
+
+  // state for managing password dialog open/close
+  const [passwordDialogOpen, setPasswordDialogOpen] = useState(false);
+
+  // generate breadcrumbs
+  const matches = useMatches();
+
+  console.log(matches);
```
Comment (mattcorner) [inline app/src/Layout.tsx:95]: Breadcrumbs are being console logged

```suggestion
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "remix-with-typescript",
-  "version": "5.0.0",
+  "name": "virtual-vehicle-builder",
```
Comment (mattcorner) [inline package.json:2]:
```suggestion
  "name": "remix-starter-ipguk",
```

**Review by mattcorner — CHANGES_REQUESTED** [review]

**Comment by mattcorner** [comment]
Tested using `npx create-remix@latest --template=file://localhost/Users/matthewcorner/Repos/remix-starter-template`.

Works well and deploys instantly.

**Comment by dmbarry86** [comment]
@evoinea you can refer to my latest commits as an example how to build for production and serve from Docker.

Note the requirement to distinguish between IAM front-end and IAM back-end as two separate environment variables. You & @mattcorner probably didn't consider this so far when running in dev mode. In production mode, the front end of our apps needs to iframe the front-end of IAM app which would be the address users would acces IAM e.g. localhost:5006. There are also some back-end calls in our apps to back-end of IAM and these require a different address for server to server.

**Comment by dmbarry86** [comment]
> Documentation link goes nowhere. We should seed with an empty docusaurus if this is our documentation tool of choice for web tools.

Agree with this. You should also include an example API resource route, api/demo etc which includes apidoc built and served a API Reference link in the sidebar.

**Comment by dmbarry86** [comment]
> You should also include an example API resource route, api/demo etc which includes apidoc built and served from a API Reference link in the sidebar.

I've done this. I'll leave one of you to do docusaurus example.

**Comment by dmbarry86** [comment]
I spotted this plugin for apidoc which lets you inject apidoc success response from a typescript interface. Could be useful. I haven't included it in my demo. https://github.com/bangbang93/apidoc-plugin-ts

**Comment by dmbarry86** [comment]
@chhack94 I've fixed the issue with doc not loading correct page when running in dev mode. I've changed the config so all of the doc is served from / rather than /help/docs as it was setup previously. The /help/docs routing should be done in the API back-end that actually serves the built doc, not in the doc config itself. I suspect this is just a carry over from the very first docusuraus project that was setup (probably NGD). We should adopt this same pattern in other apps going forward.

**Comment by chhack94** [comment]
> @chhack94 I've fixed the issue with doc not loading correct page when running in dev mode. I've changed the config so all of the doc is served from / rather than /help/docs as it was setup previously. The /help/docs routing should be done in the API back-end that actually serves the built doc, not in the doc config itself. I suspect this is just a carry over from the very first docusuraus project that was setup (probably NGD). We should adopt this same pattern in other apps going forward.

awesome, im gonna adapt this to the VVB repo too 👍

**Comment by mattcorner** [comment]
Solved issue where documentation was never actually being built and served as part of app.

**Review by mattcorner — APPROVED** [review]

**Pull request merged**